apps/studio.giselles.ai/app/(main)/tasks/[taskId]/ui/experimental/task-detail-frame.tsx

```diff
@@ -0,0 +1,184 @@
+"use client";
+
+import clsx from "clsx/lite";
+import { XIcon } from "lucide-react";
+import { AnimatePresence, motion } from "motion/react";
+import { usePathname } from "next/navigation";
+import { useEffect, useRef, useSyncExternalStore } from "react";
+import type { ImperativePanelHandle } from "react-resizable-panels";
+import { Panel, PanelGroup, PanelResizeHandle } from "react-resizable-panels";
+import { GenerationView } from "../../../../../../../../internal-packages/workflow-designer-ui/src/ui/generation-view";
+import { OutputActions } from "../output-actions";
+import { useOutputDetailPaneStore } from "./output-detail-pane-store";
+
+function useIsDesktop() {
+	return useSyncExternalStore(
+		(onStoreChange) => {
+			const mediaQueryList = window.matchMedia("(min-width: 768px)");
+			mediaQueryList.addEventListener("change", onStoreChange);
+			return () => mediaQueryList.removeEventListener("change", onStoreChange);
+		},
+		() => window.matchMedia("(min-width: 768px)").matches,
+		() => false,
+	);
+}
+
+export function TaskDetailFrame({ children }: React.PropsWithChildren) {
+	const pathname = usePathname();
+	const opened = useOutputDetailPaneStore((s) => s.opened);
+	const close = useOutputDetailPaneStore((s) => s.close);
+	const isDesktop = useIsDesktop();
+	const rightPanelRef = useRef<ImperativePanelHandle | null>(null);
+	const isPaneOpen = opened != null;
+
+	// The store can survive client-side navigation / HMR.
+	// Ensure the detail pane is closed on page enter (and when navigating between tasks).
+	// biome-ignore lint/correctness/useExhaustiveDependencies: we intentionally reset when the route changes.
+	useEffect(() => {
+		close();
+	}, [close, pathname]);
+
+	// Keep the DOM/layout stable to avoid flicker.
+	// We animate the panel size (via flex-grow transition) and fade the content in/out.
+	useEffect(() => {
+		const panel = rightPanelRef.current as unknown as {
+			collapse?: () => void;
+			expand?: () => void;
+		} | null;
+		if (!panel) return;
+		if (isPaneOpen) {
+			panel.expand?.();
+		} else {
+			panel.collapse?.();
+		}
+	}, [isPaneOpen]);
+
+	return (
+		<div className="bg-bg text-foreground h-full font-sans overflow-y-hidden">
+			{isDesktop ? (
```

Comment:
Other pages are not mobile-friendly, but is this page the only one that is?

When making the entire product mobile-friendly, we are thinking about the architecture and preparing the implementation framework, but is it necessary to implement just this page ad-hoc?